pkg/auth/inject.go

```diff
@@ -185,6 +186,42 @@ func (m DependencyMap) Provide(
 			logging.CreateLoggerWithRequestID(requestID, "provider_oauth", createLoggerMaskFormatter(tConfig)),
 			db.NewSafeTxContextWithContext(ctx, tConfig),
 		)
+	case "IdentityProvider":
+		// TODO:
+		// from tConfig
+		passwordHistoryEnabled := tConfig.UserConfig.UserAudit.Password.HistorySize > 0 || tConfig.UserConfig.UserAudit.Password.HistoryDays > 0
+		return principal.NewIdentityProvider(
+			db.NewSQLBuilder("auth", tConfig.AppName),
+			db.NewSQLExecutor(ctx, db.NewContextWithContext(ctx, tConfig)),
+			anonymous.NewSafeProvider(
```

Comment:
Should we reuse the dependent providers instead of creating them again here?

Reply:
I see many of existing nested dependencies (PasswordHistoryStore, TemplateEngine) are created individually, so I choose to follow existing convention.

Reply:
Let's change that convention. Repeating the creation is sometimes harmful because we may forget to change all of them.